iphone/Classes/TiUIWindowProxy.m

```diff
@@ -920,6 +921,45 @@ -(void)setToolbar:(id)items withObject:(id)properties
 	
 }
 
+-(void)setHidesBarsOnSwipe:(id)value
+{
+    [self replaceValue:value forKey:@"hidesBarsOnSwipe" notification:NO];
+    [self updateHidesBars];
+}
+
+-(void)setHidesBarsOnTap:(id)value
+{
+    [self replaceValue:value forKey:@"hidesBarsOnTap" notification:NO];
+    [self updateHidesBars];
+}
+
+-(void)setHidesBarsWhenVerticallyCompact:(id)value
+{
+    [self replaceValue:value forKey:@"hidesBarsWhenVerticallyCompact" notification:NO];
+    [self updateHidesBars];
+}
+
+-(void)setHidesBarsWhenKeyboardAppears:(id)value
+{
+    [self replaceValue:value forKey:@"hidesBarsWhenKeyboardAppears" notification:NO];
+    [self updateHidesBars];
+}
+
+-(void)updateHidesBars
```

Comment:
It's more about performance. Imagine, you use all provided properties. Then 5 setters will be called and 25 native calls will be made, since it calls the `[self updateHideBars]` every time. The only reason this way would make sense is, when one property depends on another one and they need to be updated as soon as another one is. If that is not the case, updating it in the setter is the way to go. 

If they require main-thread, use `ENSURE_MAIN_THREAD(methodName, argName)` to ensure that the method runs on the main-thread.